specification.markdown

```diff
@@ -173,21 +173,55 @@ An EBML Stream is a file that consists of one or many EBML Documents that are co
+
+#### EBML Schema Element Attributes
+
+Within an EBML Schema the `<element>` uses the following attributes to define an EBML Element.
+
+| attribute name | required | definition |
+|----------------|----------|------------|
+| name           | Yes      | The official human-readable name of the EBML Element. The value of the name MUST be in the form of an NCName as defined by the [XML Schema specification](http://www.w3.org/TR/1999/REC-xml-names-19990114/#ns-decl). |
+| level          | Yes      | The level notes at what hierarchical depth the EBML Element may occur within an EBML Document. The initial EBML Element of an EBML Document is at level 0 and the Elements that it may contain are at level 1. The level MUST be expressed as an integer; however, the integer may be followed by a '+' symbol to indicate that the EBML Element is allow valid at any higher level.  |
+| global         | No       | A boolean to express if an EBML Element MUST occur at its defined level be may occur within any Parent EBML Element. |
+| id             | Yes      | The Element ID expressed in hexadecimal notation prefixed by a '0x'. |
+| mandatory      | No       | A boolean to express if the EBML Element MUST occur if the Parent EBML Element is used. If the mandatory attribute is not expressed for that Element then that element is to be considered not mandatory. |
+| multiple       | No       | A boolean to express if the EBML Element may occur within its Parent EBML Element more than once. If the multiple attribute is false or the  multiple attribute is not used to define the Element then that EBML Element MUST not occur more than once with that Element's Parent EBML Element. |
+| range          | No       | For Elements which are of numerical types (Unsigned Integer, Signed Integer, Float, and Date) a numerical range may be specified. If specified that the value of the EBML Element MUST be within the defined range inclusively. |
+| default        | No       | A default value may be provided. If an Element is mandatory but not written within its Parent EBML Element, then the reader of EBML Docuemnt MUST consider the defined default value of the Element. EBML Elements that are Master-elements MUST NOT declare a default value. |
+| type           | Yes      | As defined within the (##EBML-Element-Types) section, the type MUST be set to one of the following values: 'integer' (signed integer), 'uinteger' (unsigned integer), 'float', 'string', 'date', 'utf-8', 'master', or 'binary'. |
+
+The value of the `<element>` shall contain a description that of the meaning and use of the EBML Element.
+
+#### Note on the Use of mandatory and default attributes to define EBML Elements
```

Comment:
The wording in this section conflicts with the EBML Element Types section and [how things have been done in the past](http://lists.matroska.org/pipermail/matroska-devel/2015-October/004838.html).

Consider, for example, the `CueBlockNumber` MKV element. It is a non-mandatory element with a default value of `1`. If it's encoded in the byte stream as an element with a zero-octet length, then the EBML Element Types section makes it sound like the value should be 0 ("a zero-octet length represents an integer value of zero"). There is no mention anywhere in the document that an Empty Element should use its default value.

The current wording makes it sound like only mandatory elements may have a default value, which conflicts with what is currently being done in MKV/WebM.